src/Stats.AggregateCdnDownloadsInGallery/AggregateCdnDownloadsJob.cs

```diff
@@ -147,9 +153,78 @@ public override async Task Run()
+            using (var transaction = connection.BeginTransaction(IsolationLevel.Snapshot))
+            {
+                Logger.LogInformation("Gathering Download Counts from {DataSource}/{InitialCatalog}...", connection.DataSource, connection.Database);
+
+                downloadData = (
+                    await connection.QueryWithRetryAsync<DownloadCountData>(
+                        _storedProcedureName,
+                        transaction: transaction,
+                        commandType: CommandType.StoredProcedure,
+                        commandTimeout: TimeSpan.FromSeconds(_commandTimeoutSeconds),
+                        maxRetries: 3))
+                    .ToList();
+            }
+
+            stopwatch.Stop();
+            Logger.LogInformation(
+                "Gathered {RecordCount} rows of data (took {DurationSeconds} seconds).",
+                downloadData.Count,
+                stopwatch.Elapsed.TotalSeconds);
+
+            return downloadData;
+        }
+
+        private async Task<IReadOnlyList<DownloadCountData>> GetDownloadDataFromDownloadsV1Json(string url)
+        {
+            var result = new List<DownloadCountData>();
+            var stopwatch = Stopwatch.StartNew();
+            await Retry.IncrementalAsync(async () =>
+                {
+                    Logger.LogInformation("Attempting to download {Url}", url);
+                    using (var response = await _httpClient.GetAsync(url))
+                    {
+                        response.EnsureSuccessStatusCode();
+                        using (var textReader = new StreamReader(await response.Content.ReadAsStreamAsync()))
+                        using (var jsonReader = new JsonTextReader(textReader))
+                        {
+                            DownloadsV1Reader.Load(jsonReader, (id, version, downloads) =>
+                            {
+                                result.Add(new DownloadCountData { PackageId = id, PackageVersion = version, TotalDownloadCount = downloads });
+                            });
+                        }
+                    }
+                },
+                ex => ex is HttpRequestException,
+                maxRetries: 5,
+                initialWaitInterval: TimeSpan.Zero,
+                waitIncrement: TimeSpan.FromSeconds(20));
+            stopwatch.Stop();
+
+            Logger.LogInformation("Got information about {RecordCount} packages from downloads.v1.json in {DurationSeconds} seconds",
+                result.Count,
+                stopwatch.Elapsed.TotalSeconds);
+
+            return result;
         }
 
-        private async Task ProcessBatch(List<IPackageIdGroup> batch, SqlConnection destinationDatabase, IDictionary<string, PackageRegistrationData> packageRegistrationLookup)
+        private async Task<int> ProcessBatch(List<IPackageIdGroup> batch, SqlConnection destinationDatabase, IDictionary<string, PackageRegistrationData> packageRegistrationLookup)
```

Comment:
This is existing code but ideally it would have `Async` suffix:

```suggestion
        private async Task<int> ProcessBatchAsync(List<IPackageIdGroup> batch, SqlConnection destinationDatabase, IDictionary<string, PackageRegistrationData> packageRegistrationLookup)
```